pkg/login/auth.go

```diff
@@ -39,8 +40,8 @@ func AuthenticateUser(query *models.LoginUserQuery) error {
 		return err
 	}
 
-	err := loginUsingGrafanaDB(query)
-	if err == nil || (err != models.ErrUserNotFound && err != ErrInvalidCredentials && err != ErrUserDisabled) {
+	basicAuthEnabled, err := loginUsingGrafanaDB(query)
+	if basicAuthEnabled && (err == nil || (err != models.ErrUserNotFound && err != ErrInvalidCredentials && err != ErrUserDisabled)) {
```

Comment:
It's not really basic auth. It's Grafana's built-in username/password authentication. Suggest the following

```suggestion
	loginEnabled, err := loginUsingGrafanaDB(query)
	if loginEnabled && (err == nil || (err != models.ErrUserNotFound && err != ErrInvalidCredentials && err != ErrUserDisabled)) {
```